storage/store/inmemory/unsynchronized/registers.go

```diff
@@ -0,0 +1,114 @@
+package unsynchronized
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/bits-and-blooms/bloom/v3"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type RegisterEntries map[flow.RegisterID]flow.RegisterValue
+type HeightToRegisterEntries map[uint64]RegisterEntries
+
+type Registers struct {
+	firstHeight  uint64
+	latestHeight uint64
+	lock         sync.RWMutex
+	store        HeightToRegisterEntries
+	bloomFilter  *bloom.BloomFilter
+}
+
+var _ storage.RegisterIndex = (*Registers)(nil)
+
+func NewRegisters(
+	firstHeight uint64,
+	latestHeight uint64,
+	estimatedNumberOfRegisters uint,
+	falsePositiveRate float64,
+) *Registers {
+	return &Registers{
+		firstHeight:  firstHeight,
+		latestHeight: latestHeight,
+		store:        make(HeightToRegisterEntries),
+		bloomFilter:  bloom.NewWithEstimates(estimatedNumberOfRegisters, falsePositiveRate),
+	}
+}
+
+// Get register by the register ID at a given block height.
+//
+// If the register at the given height was not indexed, returns the highest
+// height the register was indexed at.
+// Expected errors:
+// - storage.ErrHeightNotIndexed if the given height was not indexed yet or lower than the first indexed height.
+// - storage.ErrNotFound if the given height is indexed, but the register does not exist.
+func (r *Registers) Get(ID flow.RegisterID, height uint64) (flow.RegisterValue, error) {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	if height < r.firstHeight || height > r.latestHeight {
+		return flow.RegisterValue{}, storage.ErrHeightNotIndexed
+	}
+
+	if !r.bloomFilter.Test(ID.Bytes()) {
+		return flow.RegisterValue{}, storage.ErrNotFound
+	}
+
+	// Start at the requested height and go backwards
+	for h := height; h >= r.firstHeight; h-- {
```

Comment:
individual storage objects will only contain data for a single block. we don't need this logic here. There will be logic like this in the higher level abstraction.

In this method, we just need to check:
1. is height the block that was stored in this object? if not, we can return `ErrNotFound`
2. does the registerID exist in the storage object's cache?  if not, we can return `ErrNotFound`